devtools/cli/src/main/java/io/quarkus/cli/build/MavenRunner.java

```diff
@@ -191,23 +191,19 @@ public BuildCommandArgs prepareBuild(BuildOptions buildOptions, RunModeOption ru
     }
 
     @Override
-    public List<Supplier<BuildCommandArgs>> prepareDevMode(DevOptions devOptions, DebugOptions debugOptions,
-            List<String> params) {
+    public List<Supplier<BuildCommandArgs>> prepareDevTestMode(boolean devMode, DevOptions commonOptions,
+            DebugOptions debugOptions, List<String> params) {
         ArrayDeque<String> args = new ArrayDeque<>();
         List<String> jvmArgs = new ArrayList<>();
 
         setMavenProperties(args, false);
 
-        if (devOptions.clean) {
+        if (commonOptions.clean) {
             args.add("clean");
```

Comment:
If we do `clean`, it'll be safer to add `compile`, especially in a multimodule project. It may simply fail to start w/o it.

Reply:
dev will compile if something else doesn't state it. It's implied. (Test extends dev)

Reply:
If you mean it'll be called from the `dev` mojo itself then that's what I mean, it's not reliable at all especially for multimodule projects.

Reply:
hmm. That should be a separate issue then, raised against: https://github.com/quarkusio/quarkus/blob/aba7d4070845218dbee4ab95c676847320b426d2/devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java#L521

Reply:
True, it's pretty complicated though to fix it in the mojo. In general, auto-compile is more like a best effort than a reliable feature. It works pretty well for single moduled projects but not really for multimoduled ones.